### DISCUSSION TITLE
Enforcing Jump Constraint in XFEM Module

### DISCUSSION THREAD

**Discussion by pratyushpotu** [idaholab/moose#18572] [Q&A Modules: General] [answered]
Hi,

I am trying to create a model for the evaporation of liquid into vapor, and am trying to use the XFEM and Level Set modules with the LevelSetCutUserObject for the interface. I am having difficulty directly enforcing the jump in heat across the moving boundary. In particular, I am trying to use the XFEMSingleVariableConstraint to enforce this jump condition but I'm not sure if this is the best way to go. Explicitly, the condition I am trying to implement is below:

We have simulated the following equations: Let $\phi$ be the signed distance function where for each time, $t$, the zero level set is the interface between liquid and vapor phases. Explicitly, the interface between the liquid and vapor phases is given by 
![image](https://user-images.githubusercontent.com/31892913/128909563-6cb3cb72-a56e-4b03-89a3-872bc07aab0c.png)

Let the region with $\phi(x) > 0$ be the vapor phase and let the negative distances from the interface being identified with the liquid phase. Now, with this, we simulate the below two equations and reinforces energy jump condition across the interface:

![image](https://user-images.githubusercontent.com/31892913/128909602-6e9332e1-86c0-4a77-b642-6c966b315a08.png)

Any ideas/help with how to would be greatly appreciated.

Thanks,
Pratyush

**Comment by jiangwen84** [comment, marked as answer]
First, we have not really tried to combine XFEM and Level Set modules together yet. Issues there are: 1. what is the solution scheme to compute level set and physics model? Concurrently coupling or staggered coupling? 2. XFEM will cut the mesh, and you need some constraints to `heal` the mesh to run level set evolution equation. 

Second, the set of equations does not seem to be complete. I think we still need a Dirichlet like BC to enforce the `temperature jump`. So far, you only have the flux BCs on the interface. For both interfacial conditions, you can use `ElemElemConstraint` to implement your weak forms. 

Lastly, we have been developing a mesh-based moving interface capability. We use a mesh to represent the interface and we calculate the interface velocity on the cutting mesh to move the mesh. With that, you do not need to couple level set module. See #17813
